Sources/Brave/Frontend/Browser/BrowserViewController/BrowserViewController+ToolbarDelegate.swift

```diff
@@ -508,10 +510,84 @@ extension BrowserViewController: TopToolbarDelegate {
         self.present(container, animated: true)
       }
     }
+    
+    shields.showSubmitReportView = { [weak self] shieldsViewController in
+      shieldsViewController.dismiss(animated: true) {
+        guard let url = shieldsViewController.tab.url else { return }
+        self?.showSubmitReportView(for: url, isPrivateMode: shields.tab.isPrivate)
+      }
+    }
+    
     let container = PopoverNavigationController(rootViewController: shields)
     let popover = PopoverController(contentController: container, contentSizeBehavior: .preferredContentSize)
     popover.present(from: topToolbar.locationView.shieldsButton, on: self)
   }
+  
+  private func showSubmitReportView(for url: URL, isPrivateMode: Bool) {
+    // Strip fragments and query params from url
+    var components = URLComponents(url: url, resolvingAgainstBaseURL: false)
+    components?.fragment = nil
+    components?.queryItems = nil
+    guard let cleanedURL = components?.url else { return }
+    
+    let viewController = UIHostingController(rootView: SubmitReportView(
+      url: cleanedURL,
+      dismiss: { [weak self] in
+        self?.navigationController?.dismiss(animated: true)
+      },
+      submit: { [weak self] url, additionalDetails, contactInfo in
+        guard let self = self else { return }
+        self.navigationController?.dismiss(animated: true)
+        let popover = PopoverController(
+          contentController: SubmitReportSuccessViewController(), 
+          contentSizeBehavior: .preferredContentSize
+        )
```

Comment:
Why not just push a success view inside this view instead? Seems pointless to dismiss this just to show a popup

Reply:
I guess I could do that. Originally this is what I did. A success view would be definitely simple but it's just because of the size of the popover, doesn't look that nice. A large view with tiny content right in the middle, and we both agreed not to spend time trying to resize the system popover to fit the content.

Reply:
Do we have a graphic we could put in there?

Reply:
I added a big green checkmark according to the design team's suggestion